### PR TITLE
feat: make it possible to set output type, vcf or gvcf.

### PR DESCRIPTION
**Pull request by Smeds**
### Description

Make it possible to set if a vcf or gvcf will be outputted from the wrapper

### QC

* [ ] there is a test case which covers any introduced changes,
* [ ] `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* [ ] either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* [ ] rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* [ ] all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* [ ] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* [ ] all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* [ ] `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* [ ] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* [ ] the `meta.yaml` contains a link to the documentation of the respective tool or command,
* [x] `Snakefile`s pass the linting (`snakemake --lint`),
* [x] `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* [x] Python wrapper scripts are formatted with [black](https://black.readthedocs.io).

### CODE REVIEW

**Comment by Smeds** [comment]
conde refused to setup if I didn't lock mamba version to 0.15.3

File: bio/gatk/haplotypecaller/wrapper.py [hunk 1]
```diff
@@ -27,9 +27,25 @@
...
+        raise ValueError(
+            "please set vcf or gvcf as output, not both! It's not supported by gatk"
+        )
+    else:
+        raise ValueError("please set one of vcf or gvcf as output (not both)!")
+
 bam_output = snakemake.output.get("bam", "")
 if bam_output:
-    bam_output = "--bam-output " + str(bam_output)
+    output = " --bam-output " + str(bam_output)
```
Comment (fgvieira) [inline bio/gatk/haplotypecaller/wrapper.py:48]: This is an extra output, compatible with both `vcf` anf `g.vcf` ouput.

As it is, it overwrites the previous output options

**Comment by johanneskoester** [comment]
Can you revert your hotfix for the setup? I have just merged fixes to the CI config.

**Comment by Smeds** [comment]
> Can you revert your hotfix for the setup? I have just merged fixes to the CI config.

should only be changes to the wrapper left

**Review by fgvieira — APPROVED** [review]

**Pull request merged**